### PR TITLE
soft_margin_loss in pytorch frontend

### LINKED ISSUE #7592: soft_margin_loss

### PR DESCRIPTION
**Pull request by satani99**
Close #7592

### CODE REVIEW

**Comment by satani99** [comment]
Close #7592

File: ivy/functional/frontends/torch/nn/functional/loss_functions.py [hunk 1]
```diff
@@ -214,3 +214,17 @@ def nll_loss(
     ret = reduct(loss)
 
     return ret
+
+
+@to_ivy_arrays_and_back
+def soft_margin_loss(
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/torch/nn/functional/loss_functions.py:220]: The tests will fail for `float16` we need to specify unsupported dtypes
```suggestion
@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
def soft_margin_loss(
```

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @satani99 
Thanks for contributing to Ivy😊
Everything looks good just one small change is needed. check the comments

Best🙂

**Review by hirwa-nshuti — APPROVED** [review]
LGTM🙂
Thanks for contributing

**Pull request merged**